tests/integration/test_remote_write_grafana_agent.py

```diff
@@ -6,18 +6,27 @@
 import logging
 
 import pytest
-from helpers import check_prometheus_is_ready, oci_image, run_promql
+from helpers import (
+    check_prometheus_is_ready,
+    get_prometheus_rules,
+    get_rules_for,
+    oci_image,
+    run_promql,
+)
 
 logger = logging.getLogger(__name__)
 prometheus_resources = {"prometheus-image": oci_image("./metadata.yaml", "prometheus-image")}
 
 
 @pytest.mark.abort_on_fail
-async def test_remote_write_with_grafana_agent(ops_test, prometheus_charm):
+async def test_remote_write_with_grafana_agent(
+    ops_test, prometheus_charm, prometheus_tester_charm
+):
     """Test that Prometheus can be related with the Grafana Agent over remote_write."""
     prometheus_name = "prometheus"
     agent_name = "grafana-agent"
-    apps = [prometheus_name, agent_name]
+    tester_name = "prometheus-tester"
```

Comment:
To test remote write? The existing test did nothing other than "metrics work over remote write". I would hope that the comment at line 69 ("make sure topology labels are present") is a clear enough purpose. 

No other tests document the purpose of the units deployed. The logic of the test is reasonably self-documenting.

Reply:
I think we could improve clarity here: how tester's topology labels are helping with testing remote write with grafana?

Reply:
We are ensuring that label injection occurs. That's in the logic. The method name should not change because that is simply baseline functionality which is part and parcel of remote write, and the fact that a patch merge which broke this with no warning is good enough reason to have it as part of the "default" name of "test_remote_write". 

If it doesn't work, remote write is broken.